src/main/java/seedu/address/logic/commands/SellCommand.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GOOD_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_QUANTITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TRANSACTION_DATE;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.good.Good;
+import seedu.address.model.good.GoodName;
+import seedu.address.model.good.GoodQuantity;
+
+/**
+ * Sells the stated quantity of the good from the given supplier on the provided transaction date.
+ * User will not be allowed to sell more than their current stock in inventory.
+ */
+public class SellCommand extends Command {
+    public static final String COMMAND_WORD = "sell";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sells the stated "
+            + "quantity of the good on the provided transaction date. "
+            + "Users are not allowed to sell more than they have in stock. "
+            + "Parameters: "
+            + PREFIX_GOOD_NAME + "GOOD_NAME "
+            + PREFIX_QUANTITY + "QUANTITY "
+            + PREFIX_TRANSACTION_DATE + "TRANSACTION_DATE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_GOOD_NAME + "Apples "
+            + PREFIX_QUANTITY + "50 "
+            + PREFIX_TRANSACTION_DATE + "2020-01-01";
+
+    public static final String MESSAGE_SUCCESS = "Sold %1$d %2$ss";
+
+    private Good soldGood;
+
+    public SellCommand(Good soldGood) {
+        this.soldGood = soldGood;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        /*
+        if model doesn't have good, you cannot sell it
+
+        if model has good but not enough quantity, you cannot sell it
+
+        only if have good and sufficient qty then can sell
+         */
```

Comment:
I think we should keep comments written in slightly formal language and properly formatted

Reply:
Fixed.